src/extension/src/ActionHandler.py

```diff
@@ -273,10 +289,12 @@ def enable(self):
         try:
             self.setup(action=Constants.ENABLE, log_message="Enable triggered on extension")
             enable_command_handler = EnableCommandHandler(self.logger, self.telemetry_writer, self.utility, self.env_health_manager, self.runtime_context_handler, self.ext_env_handler, self.ext_config_settings_handler, self.core_state_handler, self.ext_state_handler, self.ext_output_status_handler, self.process_handler, self.cmd_exec_start_time)
-            return enable_command_handler.execute_handler_action()
+            exit_code_returned_from_executing_enable = enable_command_handler.execute_handler_action()
+            return Constants.ExitCode.Okay if exit_code_returned_from_executing_enable is None else exit_code_returned_from_executing_enable
 
         except Exception as error:
             self.logger.log_error("Error occurred during extension enable. [Error={0}]".format(repr(error)))
+            self.ext_output_status_handler.write_status_file("", self.seq_no, status=Constants.Status.Error.lower(), message="Error occurred during extension enable", code=Constants.ExitCode.HandlerFailed)
```

Comment:
I'm worried that this may overwrite a status file generated by EnableCommandHandler. Is this going to preserve the substatuses already generated?

Reply:
1. We only write substatus for "No Operation" from handler, which is preserved in this function.
2. All other substatuses are written by core, and since core runs after handler, those should be fine, even for this errored state